templates/Simple/invoice.php

```diff
@@ -56,13 +56,13 @@
 				<?php do_action( 'wpo_wcpdf_before_order_data', $this->type, $this->order ); ?>
 				<?php if ( isset($this->settings['display_number']) ) { ?>
 				<tr class="invoice-number">
-					<th><?php _e( 'Invoice Number:', 'woocommerce-pdf-invoices-packing-slips' ); ?></th>
+					<th><?php echo "{$this->get_number_title()}:"; ?></th>
```

Comment:
now that the colon is removed, does that mean we lose all the existing translations? This is a client facing string so reducing the risk of missing translations is important

Reply:
How I should address this then?

Reply:
I meant it as a question:

> does that mean we lose all the existing translations

Because I don't know if we already have these strings without colons?

Reply:
We had the colon previously.

Reply:
Yes, but that wasn't my question ;)
I just checked myself and see that we indeed already have these strings without colons:
https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/c33bde7337a08f3e1afee447e7bea5fcdbd63691/includes/documents/class-wcpdf-invoice.php#L290
https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/c33bde7337a08f3e1afee447e7bea5fcdbd63691/includes/documents/class-wcpdf-invoice.php#L275

So what remains is checking the translation coverage for these strings (they were introduced more recently than the ones with a colon), and also to check for the documents in the Professional extension.

@YordanSoares do you know if there's a quick method to check how many language packs (on wp.org) have a translation for a specific string?

Reply:
@Spreeuw there is not a public tool to search for a specific string across the different language packs of a plugin (or theme), but you can search terms within a specific locale for a plugin translation project. What I can think of is to do the following:

1. Check existing language packs [here](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/language-packs/) (currently 33).
2. Then use the following syntax to check if exist a translation for a term in each one of the packages:
[https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/**`es`**/default/?filters%5Bterm%5D=**`Invoice+Number%3A`**](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/es/default/?filters%5Bterm%5D=Invoice+Number%3A)

We need to change `es` for the locale code (e.g. `pt-bt`) and `Invoice+Number%3A` for the term to be searched.

AFAIK, Translate WordPress has an API that only allows you to get info about the current translations packages for the core, plugins and themes, but it doesn't look for translations within the packages. See [WordPress.org API > Translations](https://codex.wordpress.org/WordPress.org_API#Translations)

Right now it's a bit of a manual job, but I've already done the search for `Invoice Number:` and `Order Date:` (with  colon). If you want to search another terms, just let me know.

## Translation for `Invoice Number:` (with colon)

**Catalan (ca)**: [Nº de la factura:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/ca/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Czech (cs_CZ)**: [Číslo faktury:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/cs/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Danish (da_DK)**: [Fakturanummer:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/da/default/?filters%5Bterm%5D=Invoice+Number%3A)
**German (de_DE)**: [Rechnungsnummer:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/de/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Greek (el)**: [Αριθμός παραστατικού:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/el/default/?filters%5Bterm%5D=Invoice+Number%3A)
**English (UK) (en_GB)**: [Invoice Number:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/en-gb/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Esperanto (eo)**: [Faktura numero:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/eo/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Spanish (Colombia) (es_CO)**: [Número de factura:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/es-co/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Spanish (Venezuela) (es_VE)**: [Número de factura:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/es-ve/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Spanish (Spain) (es_ES)**: [Número de factura:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/es/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Spanish (Ecuador) (es_EC)**: [Número de factura:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/es-ec/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Spanish (Mexico) (es_MX)**: [Número de la factura:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/es-mx/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Estonian (et)**: [Arve number:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/et/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Finnish (fi)**: [Laskun numero:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/fi/default/?filters%5Bterm%5D=Invoice+Number%3A)
**French (Canada) (fr_CA)**: [Numéro de facture :](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/fr-ca/default/?filters%5Bterm%5D=Invoice+Number%3A)
**French (France) (fr_FR)**: [Numéro de la Facture :](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/fr/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Galician (gl_ES)**: [Número de factura:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/gl/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Croatian (hr)**: [Broj računa:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/hr/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Italian (it_IT)**: [Fattura N.:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/it/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Japanese (ja)**: [インボイス番号：](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/ja/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Dutch (Belgium) (nl_BE)**: [Factuurnummer:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/nl-be/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Dutch (Formal) (nl_NL_formal)**: [Factuurnummer:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/nl/formal/?filters%5Bterm%5D=Invoice+Number%3A)
**Dutch (nl_NL)**: [Factuurnummer:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/nl/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Polish (pl_PL)**: [Nr faktury:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/pl/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Portuguese (Brazil) (pt_BR)**: [Número da Fatura:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/pt-br/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Portuguese (Portugal) (pt_PT)**: [Número da factura:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/pt/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Portuguese (Portugal, AO90) (pt_PT_ao90)**: [Número da fatura:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/pt/ao90/?filters%5Bterm%5D=Invoice+Number%3A)
**Russian (ru_RU)**: [Номер счета:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/ru/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Slovak (sk_SK)**: [Číslo faktúry:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/sk/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Serbian (sr_RS)**: [Број фактуре:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/sr/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Swedish (sv_SE)**: [Fakturanummer:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/sv/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Turkish (tr_TR)**: [Fatura Numarası:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/tr/default/?filters%5Bterm%5D=Invoice+Number%3A)
**Vietnamese (vi)**: [Số hóa đơn: ](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/vi/default/?filters%5Bterm%5D=Invoice+Number%3A)

## Translation for `Order Date:` (with colon)

**Catalan (ca)**: [Data de la comanda:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/ca/default/?filters%5Bterm%5D=Order+Date%3A)
**Czech (cs_CZ)**: [Datum objednávky:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/cs/default/?filters%5Bterm%5D=Order+Date%3A)
**Danish (da_DK)**: [Ordredato:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/da/default/?filters%5Bterm%5D=Order+Date%3A)
**German (de_DE)**: [Bestelldatum:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/de/default/?filters%5Bterm%5D=Order+Date%3A)
**Greek (el)**: [Ημερομηνία παραγγελίας:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/el/default/?filters%5Bterm%5D=Order+Date%3A)
**English (UK) (en_GB)**: [Order Date:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/en-gb/default/?filters%5Bterm%5D=Order+Date%3A)
**Esperanto (eo)**: [Mendodato:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/eo/default/?filters%5Bterm%5D=Order+Date%3A)
**Spanish (Colombia) (es_CO)**: [Fecha de pedido:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/es-co/default/?filters%5Bterm%5D=Order+Date%3A)
**Spanish (Venezuela) (es_VE)**: [Fecha de pedido:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/es-ve/default/?filters%5Bterm%5D=Order+Date%3A)
**Spanish (Spain) (es_ES)**: [Fecha de pedido:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/es/default/?filters%5Bterm%5D=Order+Date%3A)
**Spanish (Ecuador) (es_EC)**: [Fecha de pedido:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/es-ec/default/?filters%5Bterm%5D=Order+Date%3A)
**Spanish (Mexico) (es_MX)**: [Fecha de pedido:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/es-mx/default/?filters%5Bterm%5D=Order+Date%3A)
**Estonian (et)**: [Tellimuse kuupäev](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/et/default/?filters%5Bterm%5D=Order+Date%3A)
**Finnish (fi)**: [Tilauksen päiväys:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/fi/default/?filters%5Bterm%5D=Order+Date%3A)
**French (Canada) (fr_CA)**: [Date de commande :](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/fr-ca/default/?filters%5Bterm%5D=Order+Date%3A)
**French (France) (fr_FR)**: [Date de commande&nbsp;:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/fr/default/?filters%5Bterm%5D=Order+Date%3A)
**Galician (gl_ES)**: [Data do pedido:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/gl/default/?filters%5Bterm%5D=Order+Date%3A)
**Croatian (hr)**: [Datum narudžbe:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/hr/default/?filters%5Bterm%5D=Order+Date%3A)
**Italian (it_IT)**: [Data Ordine:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/it/default/?filters%5Bterm%5D=Order+Date%3A)
**Japanese (ja)**: [注文日：](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/ja/default/?filters%5Bterm%5D=Order+Date%3A)
**Dutch (Belgium) (nl_BE)**: [Besteldatum:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/nl-be/default/?filters%5Bterm%5D=Order+Date%3A)
**Dutch (Formal) (nl_NL_formal)**: [Orderdatum:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/nl/formal/?filters%5Bterm%5D=Order+Date%3A)
**Dutch (nl_NL)**: [Besteldatum:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/nl/default/?filters%5Bterm%5D=Order+Date%3A)
**Polish (pl_PL)**: [Data zamówienia:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/pl/default/?filters%5Bterm%5D=Order+Date%3A)
**Portuguese (Brazil) (pt_BR)**: [Data do Pedido:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/pt-br/default/?filters%5Bterm%5D=Order+Date%3A)
**Portuguese (Portugal) (pt_PT)**: [Data da encomenda:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/pt/default/?filters%5Bterm%5D=Order+Date%3A)
**Portuguese (Portugal, AO90) (pt_PT_ao90)**: [Data da encomenda:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/pt/ao90/?filters%5Bterm%5D=Order+Date%3A)
**Russian (ru_RU)**: [Дата заказа:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/ru/default/?filters%5Bterm%5D=Order+Date%3A)
**Slovak (sk_SK)**: [Dátum objednávky:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/sk/default/?filters%5Bterm%5D=Order+Date%3A)
**Serbian (sr_RS)**: [Датум поруџбине:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/sr/default/?filters%5Bterm%5D=Order+Date%3A)
**Swedish (sv_SE)**: [Orderdatum:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/sv/default/?filters%5Bterm%5D=Order+Date%3A)
**Turkish (tr_TR)**: [Sipariş Tarihi:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/tr/default/?filters%5Bterm%5D=Order+Date%3A)
**Vietnamese (vi)**: [Ngày đặt hàng:](https://translate.wordpress.org/projects/wp-plugins/woocommerce-pdf-invoices-packing-slips/stable/vi/default/?filters%5Bterm%5D=Order+Date%3A)